transform_encoder.py

```diff
@@ -0,0 +1,131 @@
+        self.embedding_fn_name = embedding_fn_name
+        self.tokenizer = AutoTokenizer.from_pretrained(self.base_tokenizer_model)
+        self.model = AutoModel.from_pretrained(
+            self.pretrained_model_name_or_path, output_hidden_states=True
+        )
+        self.model.to(torch.device(device))
+
+    def _compute_embedding(
+        self, hidden_states: Tuple["torch.Tensor"], input_tokens: Dict
+    ):
+        fill_vals = {"cls": 0.0, "mean": 0.0, "max": -np.inf, "min": np.inf}
+        fill_val = torch.tensor(
+            fill_vals[self.pooling_strategy], device=torch.device(self.device)
+        )
+        layer = hidden_states[self.layer_index]
+        attn_mask = input_tokens["attention_mask"].unsqueeze(-1).expand_as(layer)
+        layer = torch.where(attn_mask.bool(), layer, fill_val)
+        embeddings = layer.sum(dim=1) / attn_mask.sum(dim=1)
+        return embeddings.cpu().numpy()
+
+    @requests
+    def encode(self, docs: "DocumentArray", parameters: Dict, **kwargs):
+        for batch in self._get_docs_batch_generator(docs, parameters):
+            texts = batch.get_attributes("text")
+
+            with torch.no_grad():
+                input_tokens = self._generate_input_tokens(texts)
+                outputs = getattr(self.model, self.embedding_fn_name)(**input_tokens)
+                if isinstance(outputs, torch.Tensor):
+                    outputs = outputs.cpu().numpy()
+                hidden_states = outputs.hidden_states
+                embeds = self._compute_embedding(hidden_states, input_tokens)
+                for doc, embed in zip(batch, embeds):
+                    doc.embedding = embed
+
+    def _generate_input_tokens(self, texts):
+        if not self.tokenizer.pad_token:
+            self.tokenizer.add_special_tokens({"pad_token": "[PAD]"})
+            self.model.resize_token_embeddings(len(self.tokenizer.vocab))
+
+        input_tokens = self.tokenizer(
+            texts,
+            max_length=self.max_length,
+            padding="longest",
+            truncation=True,
+            return_tensors="pt",
+        )
+        input_tokens = {
+            k: v.to(torch.device(self.device)) for k, v in input_tokens.items()
+        }
+        return input_tokens
+
+    def _get_docs_batch_generator(self, docs: DocumentArray, parameters: Dict):
+        traversal_path = parameters.get("traversal_path", self.default_traversal_path)
+        batch_size = parameters.get("batch_size", self.default_batch_size)
+        flat_docs = docs.traverse_flat(traversal_path)
+        filtered_docs = DocumentArray(
+            [doc for doc in flat_docs if doc is not None and doc.text is not None]
+        )
+        return _batch_generator(filtered_docs, batch_size)
+
+
+def _batch_generator(
+    data: DocumentArray, batch_size: int
+) -> Generator[DocumentArray, None, None]:
+    for i in range(0, len(data), batch_size):
+        yield data[i : min(i + batch_size, len(data))]
```

Comment:
u do not need to compute the min here as @davidbp is explaining in different PRs

Reply:
https://github.com/jina-ai/executor-transformer-torch-encoder/pull/1/commits/9116b7c45458db6495de5eca7b8f2d31dd4104c1